content/how-to/hot-module-reload.md

```diff
@@ -0,0 +1,186 @@
+---
+title: How to Configure Hot Module Replacement?
+---
+Hot Module Replacement (HMR) exchanges, adds, or removes modules while an application is running without a page reload. HMR is particularly useful in applications using a single state tree, since components are "dumb" and will reflect the latest application state, even after their source is changed and they are replaced. 
+
+Webpack's power lies in its customizablity, and there are MANY ways of configuring HMR given the needs of a partiular project. The approach described below uses Babel and React, but these tools are not necessary for HMR to work. If you'd like to see examples of other approaches, please request them or, better yet, open up a PR with an addition! 
+
+##Project Config
+To use HMR, you'll need the following dependencies:
+
+* webpack 2 (tested with 2.1.0-beta.20)
+* webpack-dev-server 2 (tested with 2.1.0-beta.0)
+
+This guide will be demonstrating the use of HMR with Babel, React, and PostCSS (using CSS Modules). To follow along, please add the following deps to your `package.json`:
+
+```js
+"devDependencies": {
+    "babel": "^6.5.2",
+    "babel-core": "^6.11.4",
+    "babel-loader": "^6.2.4",
+    "babel-preset-es2015-webpack": "^6.4.2",
+    "babel-preset-react": "^6.11.1",
+    "babel-preset-stage-2": "^6.11.0",
+    "css-loader": "^0.23.1",
+    "postcss-loader": "^0.9.1",
+    "react-hot-loader": "^3.0.0-beta.1",
+    "style-loader": "^0.13.1",
+    "webpack": "^2.1.0-beta.20",
+    "webpack-dev-server": "^2.1.0-beta.0"
+  }
+```
+
+###Babel Config
+Your `.babelrc` file should look like the following:
+
+```js
+{
+  "presets": ["es2015-webpack", "stage-2", "react"],
+  "plugins": ["react-hot-loader/babel"]
+}
+```
+
+There's a few things to note here. First, in the presets, we're using es2015-webpack, which is a Babel preset that enables support for es2015 while DISABLING conversion to CommonJS module format. Since Webpack 2 understands import/export syntax, the es2015-webpack preset (unlike the general es2015 preset) allows Webpack to perform tree-shaking and dead code elimination.
+
+The stage-2 preset is up to your comfort with adopting new technoligies; stage-0 will keep you on the bleeding age.
```

Comment:
I would add a warning (`W>`) about using Stage 0 specifically as it can potentially break your code. I prefer to enable features on feature basis over stages to communicate better which custom features I'm using. Better for maintenance.